src/lib/ast.rs

```diff
@@ -762,6 +750,15 @@ impl<U: PartialEq + Copy> NodeId<U> {
                    .position(|val| val == self)
     }
 
+    /// Return the id of the first leaf of the subtree rooted on this node, if there is one.
+    pub fn get_first_leaf<T: Clone>(self, arena: &Arena<T, U>) -> NodeId<U> {
+        let mut node = self;
+        while !node.is_leaf(arena) {
+            node = node.children(&arena).collect::<Vec<NodeId<U>>>()[0]
```

Comment:
If the first `node.is_lef()` call is `false`, doing a `collect` here is going to be pretty inefficient: it has to malloc memory, copy things over etc. etc. Can we not do a version that simply references things rather than copying?